tests/System/Routes/AddDonationRouteTest.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace WMDE\Fundraising\Frontend\Tests\System\Routes;
+
+use WMDE\Fundraising\Frontend\FunFunFactory;
+use WMDE\Fundraising\Frontend\Tests\Fixtures\DonationRepositorySpy;
+use WMDE\Fundraising\Frontend\Tests\System\WebRouteTestCase;
+
+/**
+ * @licence GNU GPL v2+
+ * @author Kai Nissen < kai.nissen@wikimedia.de >
+ */
+class AddDonationRouteTest extends WebRouteTestCase {
+
+	private $validFormInput = [
+		'betrag' => '5,51',
+		'zahlweise' => 'BEZ',
+		'periode' => 0,
+		'iban' => 'DE12500105170648489890',
+		'bic' => 'INGDDEFFXXX',
+		'konto' => '0648489890',
+		'blz' => '50010517',
+		'bankname' => 'ING-DiBa',
+		'adresstyp' => 'person',
+		'anrede' => 'Frau',
+		'titel' => 'Prof. Dr.',
+		'firma' => '',
+		'vorname' => 'Karla',
+		'nachname' => 'Kennichnich',
+		'strasse' => 'Lehmgasse 12',
+		'plz' => '12345',
+		'ort' => 'Einort',
+		'country' => 'DE',
+		'email' => 'karla@kennichnich.de',
+		'info' => '1',
+		'piwik_campaign' => 'test',
+		'piwik_kwd' => 'gelb',
+		'impCount' => '3',
+		'bImpCount' => '1',
+		'layout' => 'Default',
+		'color' => 'blue',
+		'skin' => 'default',
+	];
+
+	public function testGivenValidRequest_donationGetsPersisted() {
+		$donationRepository = new DonationRepositorySpy();
+
+		$client = $this->createClient( [], function( FunFunFactory $factory ) use ( $donationRepository ) {
+			$factory->setDonationRepository( $donationRepository );
```

Comment:
The responsibility of the system tests is to verify the whole stack works, so it's generally better to not mock out the database. Changed in #183